clients/cobol-lsp-vscode-extension/src/__tests__/services/copybook/E4ECopybookService.spec.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2024 Broadcom.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Broadcom, Inc. - initial API and implementation
+ */
+import * as vscode from "vscode";
+import { E4ECopybookService } from "../../../services/copybook/E4ECopybookService";
+import { e4eMock } from "./download/getE4EMock.utility";
+import { Utils } from "../../../services/util/Utils";
+
+describe("e4e copybook service tests", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+  it("checks valid e4e api retrieved", async () => {
+    const ext = { exports: { ...e4eMock }, activate: jest.fn() };
+    vscode.extensions.getExtension = jest.fn().mockReturnValue(ext);
+
+    const spyValidate = jest.spyOn(Utils, "validateE4E");
+    const api = await E4ECopybookService.getE4EAPI();
+    expect(spyValidate).toHaveBeenCalled();
+    expect(api === e4eMock);
```

Comment:
I don't think this does anything, because `api` and `e4eMock` are separate objects, so the expression is `false`

Reply:
Testing if validation is called and retrieved the api correctly. get extension should be mocked anyway.

Reply:
This statement `expect(api === e4eMock);` literally does not do anything - there is no toBeTruthy, or toBeEqual etc.